server/metrics/metrics.go

```diff
@@ -9,9 +23,25 @@ import (
 	promreporter "github.com/uber-go/tally/prometheus"
 )
 
+type GrpcCounter struct {
+	Name    string
+	Tags    map[string]string
+	Counter tally.Counter
+}
+
+type GrpcHistogram struct {
+	Name      string
+	Tags      map[string]string
+	Histogram tally.Histogram
+}
+
 var (
 	Root     tally.Scope
 	Reporter promreporter.Reporter
+	// method name and counter name
+	GrpcCounters map[string]map[string]*GrpcCounter
+	// method name and histogram name
+	GrpcHistograms map[string]map[string]*GrpcHistogram
```

Comment:
nit: also add a comment on these maps that these are only initialized once during startup to avoid the need for synchronization.

Reply:
Good point, added